### PR TITLE
[Notebook] Embed preview action fix

### LINKED ISSUE #2842: [Notebook] Previewing snaphots or embeds throws error
This is the error seen: 

vue.js?1531:580 [Vue warn]: Error in mounted hook: "TypeError: Cannot read property 'key' of undefined"

found in

---> <Preview>
       <Root>
le @ vue.js?1531:580
Ge @ vue.js?1531:1698
Ve @ vue.js?1531:1693
Xe @ vue.js?1531:1682
jt @ vue.js?1531:2874
insert @ vue.js?1531:4043
O @ vue.js?1531:5792
eval @ vue.js?1531:6011
e._update @ vue.js?1531:2617
i @ vue.js?1531:2745
Gt.get @ vue.js?1531:3086
Gt @ vue.js?1531:3075
eval @ vue.js?1531:2749
On.$mount @ vue.js?1531:8317
On.$mount @ vue.js?1531:10641
invoke @ PreviewAction.js?92d6:51
perform @ notebook-embed.vue?c381:227
click @ notebook-embed.vue?1c34:1
t @ vue.js?1531:1989
t._withTask.o._withTask @ vue.js?1531:1787
vue.js?1531:1702 TypeError: Cannot read property 'key' of undefined
    at setView (Preview.vue?6664:122)
    at s.n [as setView] (vue.js?1531:194)
    at s.mounted (Preview.vue?6664:96)
    at jt (vue.js?1531:2872)
    at Object.insert (vue.js?1531:4043)
    at O (vue.js?1531:5792)
    at On.eval [as __patch__] (vue.js?1531:6011)
    at On.e._update (vue.js?1531:2617)
    at On.i (vue.js?1531:2745)
    at Gt.get (vue.js?1531:3086)
Ge @ vue.js?1531:1702
Ve @ vue.js?1531:1693
Xe @ vue.js?1531:1682
jt @ vue.js?1531:2874
insert @ vue.js?1531:4043
O @ vue.js?1531:5792
eval @ vue.js?1531:6011
e._update @ vue.js?1531:2617
i @ vue.js?1531:2745
Gt.get @ vue.js?1531:3086
Gt @ vue.js?1531:3075
eval @ vue.js?1531:2749
On.$mount @ vue.js?1531:8317
On.$mount @ vue.js?1531:10641
invoke @ PreviewAction.js?92d6:51
perform @ notebook-embed.vue?c381:227
click @ notebook-embed.vue?1c34:1
t @ vue.js?1531:1989
t._withTask.o._withTask @ vue.js?1531:1787

### PR DESCRIPTION
**Pull request by deeptailor**
pass correct object path, and remove notebook snapshot button from preview window

Fixes #2842

### CODE REVIEW

File: src/ui/preview/Preview.vue [hunk 1]
```diff
@@ -40,12 +40,6 @@
                     :current-view="currentView"
                     @setView="setView"
                 />
```
Comment (nikhilmandlik) [inline src/ui/preview/Preview.vue:42]: Thanks for removing this, I am also working on it and tracked in https://github.com/nasa/openmct/issues/2821

File: src/plugins/notebook/components/notebook-embed.vue [hunk 2]
```diff
@@ -217,18 +217,16 @@ export default {
...
-                        self.actions.push({
-                            cssClass: action.cssClass,
-                            name: action.name,
-                            perform: () => {
-                                action.invoke([domainObject].concat(self.openmct.router.path));
-                            }
-                        });
-                    });
+
+            actions.forEach((action) => {
```
Comment (nikhilmandlik) [inline src/plugins/notebook/components/notebook-embed.vue:221]: Thanks for changes, Andrew commented on this in Notebook PR as TODO items. :)

File: src/plugins/notebook/utils/notebook-entries.js [hunk 3]
```diff
@@ -95,7 +95,8 @@ export const createNewEmbed = (snapshotMeta, snapshot = '') => {
         id: 'embed-' + date,
         name,
         snapshot,
-        type
+        type,
+        objectPath
```
Comment (nikhilmandlik) [inline src/plugins/notebook/utils/notebook-entries.js:99]: adding this causing error,
`Error in event handler for "updateEntries": "TypeError: Converting circular structure to JSON`

may be can do something like, `objectPath: JSON.stringify(objectPath)` then retrieve with `JSON.parse(self.embed.objectPath)`

Also worried about storing this objectPath info in each embed, overall performance for fetch, sort and search.
Let's discuss this tomorrow?

Comment (deeptailor) [reply src/plugins/notebook/utils/notebook-entries.js]: In order for preview to work, we have to give it the correct objectPath. Using the url objectPath will cause all kinds of issues with other context menu actions in preview.

I turned it back to json before passing it on.

**Pull request merged**